pyuvdata/uvcal/uvcal.py

```diff
@@ -3327,6 +3328,472 @@ def __iadd__(
+                "_time_range",
+            ]
+        elif axis == "jones":
+            history_update_string += "jones"
+            compatibility_params += [
+                "_freq_array",
+                "_channel_width",
+                "_ant_array",
+                "_time_array",
+                "_integration_time",
+                "_lst_array",
+                "_time_range",
+            ]
+        elif axis == "antenna":
+            history_update_string += "antenna"
+            compatibility_params += [
+                "_freq_array",
+                "_channel_width",
+                "_jones_array",
+                "_time_array",
+                "_integration_time",
+                "_lst_array",
+                "_time_range",
+            ]
+        elif axis == "time":
+            history_update_string += "time"
+            compatibility_params += [
+                "_freq_array",
+                "_channel_width",
+                "_jones_array",
+                "_ant_array",
+            ]
+            if not this.future_array_shapes:
+                compatibility_params += ["_integration_time"]
+
+        history_update_string += " axis using pyuvdata."
+        histories_match = []
+        for obj in other:
+            histories_match.append(uvutils._check_histories(this.history, obj.history))
+
+        this.history += history_update_string
+        for obj_num, obj in enumerate(other):
+            if not histories_match[obj_num]:
+                if verbose_history:
+                    this.history += " Next object history follows. " + obj.history
+                else:
+                    extra_history = uvutils._combine_history_addition(
+                        this.history, obj.history
+                    )
+                    if extra_history is not None:
+                        this.history += (
+                            " Unique part of next object history follows. "
+                            + extra_history
+                        )
+        # Actually check compatibility parameters
+        for obj in other:
+            for a in compatibility_params:
+                params_match = getattr(this, a) == getattr(obj, a)
+                if not params_match:
+                    msg = (
+                        "UVParameter "
+                        + a[1:]
+                        + " does not match. Cannot combine objects."
+                    )
+                    raise ValueError(msg)
+
+            for a in warning_params:
+                params_match = getattr(this, a) == getattr(obj, a)
+                if not params_match:
+                    msg = "UVParameter " + a[1:] + " does not match. Combining anyway."
+                    warnings.warn(msg)
+
+        total_quality_exists = [this.total_quality_array is not None] + [
+            obj.total_quality_array is not None for obj in other
+        ]
+        this_tqa_exp_shape = this._total_quality_array.expected_shape(this)
```

Comment:
It looks like this just serves as the "default" shape to fill in for objects that don't have a `total_quality_array`. Do we do a check (explicitly or implicitly) that it's the "right" shape? Is that handled as part of the `check` calls to the individual objects in `other` above?

Reply:
I think it's handled by the check, but I'll look at it carefully and see if I can make it any better.